src/blib2to3/pgen2/driver.py

```diff
@@ -194,12 +194,26 @@ def load_grammar(
     if force or not _newer(gp, gt):
         logger.info("Generating grammar tables from %s", gt)
         g: grammar.Grammar = pgen.generate_grammar(gt)
-        if save:
+        if save and os.path.exists(os.path.dirname(gp)):
```

Comment:
We should do this only in an except block because the directory may be deleted between the time we run `os.path.exists` and the time we actually write to the file.

Reply:
That is a good point. Using an except block still has the look of an bad error though. Given that I prefer if the cache directory doesn't exist an informative message should be shown instead of creating the directory, I'll change the PR to improve the output to be more informative instead of skipping.

Reply:
It is starting to look like greedily creating the parent directories of the cache directory looks like a better idea though. Even with an output like:
```
INFO:blib2to3.pgen2.driver:Generating grammar tables from /home/richard-26/programming/black/src/blib2to3/Grammar.txt
INFO:blib2to3.pgen2.driver:Writing grammar tables to /home/richard-26/.cache/black/19.10b1.dev122+gf90f50a/Grammar3.8.1.final.0.pickle
INFO:blib2to3.pgen2.driver:Writing failed because the directory where the grammar tables would be saved to is non-existent  
``` 

still looks like an error, causing more issues to be opened for it. The root problem is that when using Black as a library, the cache directory isn't created since most folks call `reformat_one` directly. I'm not sure how to proceed since the problem I'm trying to fix is people seeing the info log and opening more issues. Like https://github.com/nteract/papermill/issues/498, https://github.com/wpbrown/azmeta-codespace/issues/2, and https://github.com/psf/black/issues/1464.